src/main/java/com/microsoft/azure/springcloudplayground/controller/MainController.java

```diff
@@ -75,22 +101,52 @@ private void addBuildInformation(@NonNull Map<String, Object> model) {
         model.put("build.information", buildInfo);
     }
 
+    private void triggerGenerateEvent(@NonNull List<String> services, @NonNull String ipAddress) {
+        final Map<String, String> properties = new HashMap<>();
+
+        services.forEach(s -> properties.put(s, "selected"));
+        properties.put("accessIp", ipAddress);
+
+        this.telemetryProxy.trackEvent(TELEMETRY_EVENT_GENERATE, properties);
+    }
+
+    private void triggerAccessEvent(@NonNull String ipAddress) {
+        final Map<String, String> properties = new HashMap<>();
+
+        properties.put("accessIp", ipAddress);
+
+        this.telemetryProxy.trackEvent(TELEMETRY_EVENT_ACCESS, properties);
+    }
+
+    private void triggerLoginEvent(@NonNull String accountType, @NonNull String ipAddress) {
+        final Map<String, String> properties = new HashMap<>();
+
+        properties.put("accountType", accountType);
+        properties.put("accessIp", ipAddress);
+
+        this.telemetryProxy.trackEvent(TELEMETRY_EVENT_LOGIN, properties);
+    }
+
     @RequestMapping(path = "/", produces = "text/html")
-    public String home(Map<String, Object> model) {
+    public String home(Map<String, Object> model, HttpServletRequest request) {
+
         this.addBuildInformation(model);
         this.renderHome(model);
+        this.triggerAccessEvent(request.getRemoteAddr());
 
         return "home";
     }
 
     @RequestMapping("/starter.zip")
```

Comment:
change /starter.zip to something else e.g. /microservice.zip?

Reply:
copy that.